go/vt/vtgate/vindexes/lookup_internal.go

```diff
@@ -73,21 +73,87 @@ func (lkp *lookupInternal) Lookup(vcursor VCursor, ids []sqltypes.Value) ([]*sql
 		return nil, fmt.Errorf("cannot perform lookup: no vcursor provided")
 	}
 	results := make([]*sqltypes.Result, 0, len(ids))
-	for _, id := range ids {
+	if !ids[0].IsIntegral() && !ids[0].IsBinary() {
+		// for non integral and binary type, fallback to send query per id
+		for _, id := range ids {
+			vars, err := sqltypes.BuildBindVariable([]interface{}{id})
+			if err != nil {
+				return nil, fmt.Errorf("lookup.Map: %v", err)
+			}
+			bindVars := map[string]*querypb.BindVariable{
+				lkp.FromColumns[0]: vars,
+			}
+			var result *sqltypes.Result
+			co := vtgatepb.CommitOrder_NORMAL
+			if lkp.Autocommit {
+				co = vtgatepb.CommitOrder_AUTOCOMMIT
+			}
+			result, err = vcursor.Execute("VindexLookup", lkp.sel, bindVars, false /* rollbackOnError */, co)
+			if err != nil {
+				return nil, fmt.Errorf("lookup.Map: %v", err)
+			}
+			rows := make([][]sqltypes.Value, 0, len(result.Rows))
+			for _, row := range result.Rows {
+				rows = append(rows, []sqltypes.Value{row[1]})
+			}
+			results = append(results, &sqltypes.Result{
+				Rows: rows,
+			})
+		}
+	} else {
+		// for integral or binary type, batch query all ids and then map them back to the input order
+		idVars := make([]interface{}, len(ids))
+		for i := range ids {
+			idVars[i] = ids[i]
+		}
+		vars, err := sqltypes.BuildBindVariable(idVars)
```

Comment:
I'm surprised that we don't support `[]Value` for building bind variables. Can you extend `BuildBindVariable` instead? That will save us the extra allocation of `idVars`.